ext/storage_sources/local_store/local_store.c

```diff
@@ -0,0 +1,1220 @@
+    len = v->len + strlen(suffix) + 1;
+    if ((p = malloc(len)) == NULL)
+        return (local_err(local, session, ENOMEM, "configuration parsing"));
+    (void)snprintf(p, len, "%.*s", (int)v->len, v->str);
+
+    /*
+     * Check for illegal characters before adding the suffix, as the suffix may contain such
+     * characters.
+     */
+    if (disallowed != NULL && strstr(p, disallowed) != NULL) {
+        ret = local_err(local, session, EINVAL,
+          "characters \"%s\" disallowed in configuration string \"%s\"", disallowed, p);
+        free(p);
+        return (ret);
+    }
+    (void)strncat(p, suffix, len);
+    *result = p;
+    return (0);
+}
+
+/*
+ * local_configure
+ *     Parse the configuration for the keys we care about.
+ */
+static int
+local_configure(LOCAL_STORAGE *local, WT_CONFIG_ARG *config)
+{
+    int ret;
+
+    if ((ret = local_configure_int(local, config, "force_error", &local->force_error)) != 0)
+        return (ret);
+    if ((ret = local_configure_int(local, config, "force_delay", &local->force_delay)) != 0)
+        return (ret);
+    if ((ret = local_configure_int(local, config, "delay_ms", &local->delay_ms)) != 0)
+        return (ret);
+    if ((ret = local_configure_int(local, config, "verbose", &local->verbose)) != 0)
+        return (ret);
+
+    return (0);
+}
+
+/*
+ * local_configure_int
+ *     Look for a particular configuration key, and return its integer value.
+ */
+static int
+local_configure_int(LOCAL_STORAGE *local, WT_CONFIG_ARG *config, const char *key, uint32_t *valuep)
+{
+    WT_CONFIG_ITEM v;
+    int ret;
+
+    ret = 0;
+
+    if ((ret = local->wt_api->config_get(local->wt_api, NULL, config, key, &v)) == 0) {
+        if (v.len == 0 || v.type != WT_CONFIG_ITEM_NUM)
+            ret = local_err(local, NULL, EINVAL, "force_error config arg: integer required");
+        else
+            *valuep = (uint32_t)v.val;
+    } else if (ret == WT_NOTFOUND)
+        ret = 0;
+    else
+        ret = local_err(local, NULL, EINVAL, "WT_API->config_get");
+
+    return (ret);
+}
+
+/*
+ * local_err --
+ *     Print errors from the interface. Returns "ret", the third argument.
+ */
+static int
+local_err(LOCAL_STORAGE *local, WT_SESSION *session, int ret, const char *format, ...)
+{
+    va_list ap;
+    WT_EXTENSION_API *wt_api;
+    char buf[1000];
+
+    va_start(ap, format);
+    wt_api = local->wt_api;
+    if (vsnprintf(buf, sizeof(buf), format, ap) > (int)sizeof(buf))
+        wt_api->err_printf(wt_api, session, "local_storage: error overflow");
+    wt_api->err_printf(
+      wt_api, session, "local_storage: %s: %s", wt_api->strerror(wt_api, session, ret), buf);
+    va_end(ap);
+
+    return (ret);
+}
+
+/*
+ * local_flush_free --
+ *     Free storage for a flush item.
+ */
+static void
+local_flush_free(LOCAL_FLUSH_ITEM *flush)
+{
+    if (flush != NULL) {
+        free(flush->src_path);
+        free(flush->marker_path);
+        free(flush->bucket);
+        free(flush->kmsid);
```

Comment:
Alphabetize these as I don't think the order is dependent on which is freed first.